File: 2363.Merge-Similar-Items.js
/**
 * @param {number[][]} items1
 * @param {number[][]} items2
 * @return {number[][]}
 */
var mergeSimilarItems = function (items1, items2) {
  const map = new Map();

  for (const [key, val] of items1) {
    map.set(key, (map.get(key) || 0) + val);
  }

  for (const [key, val] of items2) {
    map.set(key, (map.get(key) || 0) + val);
  }

  return [...map.entries()].sort((a, b) => a[0] - b[0]);
};

File: python/2363.Merge-Similar-Items.py
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
        items = {}

        for key, val in items1:
            items[key] = items.get(key, 0) + val

        for key, val in items2:
            items[key] = items.get(key, 0) + val

        return [[k, v] for k, v in sorted(items.items())]